packages/node_modules/@webex/test-users/src/index.js

```diff
@@ -0,0 +1,290 @@
+ */
+function getClientCredentials({clientId, clientSecret, idbrokerUrl}) {
+  if (clientToken) {
+    return Promise.resolve(clientToken);
+  }
+
+  return request({
+    method: 'POST',
+    uri: `${idbrokerUrl}/idb/oauth2/v1/access_token`,
+    json: true,
+    form: {
+      grant_type: 'client_credentials',
+      scope: 'webexsquare:get_conversation',
+      client_id: clientId,
+      client_secret: clientSecret
+    },
+    headers: {
+      // Note: we can't request's auth hash here because this endpoint expects
+      // us to send the auth header *without including "Basic "* before the
+      // token string
+      authorization: btoa(`${clientId}:${clientSecret}`)
+    }
+  })
+    .then((res) => {
+      const token = fixToken(res.body);
+
+      return `${token.token_type} ${token.access_token}`;
+    })
+    .then((token) => {
+      clientToken = token;
+
+      return clientToken;
+    });
+}
+
+/**
+ * Makes a request authorized with client credentials
+ * @param {Object} options
+ * @param {Object} options.body
+ * @param {string} options.body.clientId
+ * @param {string} options.body.clientSecret
+ * @param {string} options.body.idbrokerUrl
+ * @private
+ * @returns {Promise<HttpResponseObject>}
+ */
+function requestWithAuth(options) {
+  return getClientCredentials(options.body)
+    .then((authorization) => {
+      options.headers = options.headers || {};
+      options.headers.authorization = authorization;
+
+      return request(options);
+    });
+}
+
+/**
+ * @typedef {Object} AccessTokenObject
+ * @property {string} token.access_token
+ * @property {Number} token.expires_in
+ * @property {string} token.token_type
+ * @property {string} token.refresh_token
+ * @property {Number} token.refresh_token_expires_in
+ * @property {string} token.expires
+ * @property {string} token.refresh_token_expires
+ */
+
+/**
+ * @typedef {Object} CreateUserOptions
+ * @property {string} [displayName]
+ * @param {string} [clientId] defaults to WEBEX_CLIENT_ID
+ * @param {string} [clientSecret] defaults to WEBEX_CLIENT_SECRET
+ * @param {string} [idbrokerUrl] defaults to IDBROKER_BASE_URL
+ * @param {string} [conversationServiceUrl] defaults to WEBEX_TEST_USERS_CONVERSATION_SERVICE_URL
+ * @property {string} [emailAddress]
+ * @property {Array.<string>} entitlements
+ * @property {string} [scope] defaults to WEBEX_SCOPE
+ * @property {boolean} authCodeOnly generates auth_code
+ */
+
+/**
+  * @typedef {Object} TestUserObject
+  * @property {string} password
+  * @property {string} emailAddress
+  * @property {string} displayName
+  * @property {string} id
+  * @property {string} userName
+  * @property {string} email
+  * @property {string} name
+  * @property {string} givenName
+  * @property {string} type
+  * @property {Array.<string>} entitlements
+  * @property {string} orgId
+  * @property {AccessTokenObject} token
+  */
+
+/**
+ * Creates a test user
+ * @param {CreateUserOptions} options
+ * @returns {Promise.<TestUserObject>}
```

Comment:
In relation to previous comments, should we only create `@typedef` references on larger objects as displayed here?

Reply:
We should definitely use `@typedef` for large objects on public functions. For some of the private functions that received the options object but only used a small subset of the params, I simply defined the params the function used.